src/ralph/dns/views.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+import logging
+from urllib.parse import urlencode
+
+import requests
+from django.conf import settings
+from django.contrib import messages
+from django.forms import BaseFormSet, formset_factory
+from django.http import HttpResponseRedirect
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.admin.views.extra import RalphDetailView
+from ralph.dns.forms import DNSRecordForm, RecordType
+
+logger = logging.getLogger(__name__)
+
+
+def get_api_kwargs(url, data=None, headers=None):
+    if headers is None:
+        headers = {}
+    headers.update({'Authorization': 'Token {}'.format(settings.DNSAAS_TOKEN)})
+
+    return {'url': url, 'data': data, 'headers': headers}
+
+
+def get_api_result(url):
+    request = requests.get(**get_api_kwargs(url))
+    json_data = request.json()
+    api_results = json_data.get('results', [])
+    if json_data.get('next', None):
+        _api_results = get_api_result(json_data['next'])
+        api_results.extend(_api_results)
+    return api_results
+
+
+def get_dns_records(ipaddresses):
+    """Gets DNS Records for `ipaddresses` by API call"""
+    dns_records = []
+    ipaddresses = [('ip', i) for i in ipaddresses]
+    url = '{}/{}/?{}'.format(
+        settings.DNSAAS_URL,
+        'api/records',
+        urlencode([
+            ('limit', 100),
+            ('offset', 0)
+        ] + ipaddresses)
+    )
+    api_results = get_api_result(url)
+    ptr_list = set([i['content'] for i in api_results if i['type'] == 'PTR'])
+
+    for item in api_results:
+        if item['type'] in {'A', 'CNAME', 'TXT'}:
+            dns_records.append({
+                'pk': item['id'],
+                'name': item['name'],
+                'type': RecordType.from_name(item['type'].lower()).id,
+                'content': item['content'],
+                'ptr': item['name'] in ptr_list
+            })
+    return dns_records
+
+
+def update_dns_records(records):
+    error = False
+    for item in records:
+        url = '/'.join([settings.DNSAAS_URL, 'api/records', str(item['pk'])])
```

Comment:
IMO it will (i see 301, not 302, but it doesn't make any difference here):
```
curl -v https://<DNSAAS_URL>/api/records/1
...
< HTTP/1.1 301 MOVED PERMANENTLY
...
< Location: https://<DNSAAS_URL>/api/records/1/
```

I'd add `/` at the end and we'll make twice less requests.